kafka-connect/src/main/java/io/tabular/iceberg/connect/channel/Coordinator.java

```diff
@@ -255,6 +288,77 @@ private void commitToTable(
     }
   }
 
+  private static String dataFilePath(DataFile dataFile) {
+    return dataFile.path().toString();
+  }
+
+  private static String deleteFilePath(DeleteFile deleteFile) {
+    return deleteFile.path().toString();
+  }
+
+  private <T> List<T> deduplicateBatch(
+      List<T> files,
+      Function<T, String> getPathFn,
+      String fileType,
+      UUID commitId,
+      TableIdentifier tableIdentifier) {
+    return deduplicate(
+        files,
+        getPathFn,
+        (numDuplicatedFiles, path) ->
+            String.format(
+                "Detected %d %s files with the same path=%s across payloads during commitId=%s for table=%s",
+                numDuplicatedFiles,
+                fileType,
+                path,
+                commitId.toString(),
+                tableIdentifier.toString()));
+  }
+
+  private <T> List<T> deduplicatePayload(
+      List<T> files,
+      Function<T, String> getPathFn,
+      String fileType,
+      UUID commitId,
+      TableIdentifier tableIdentifier) {
+    return deduplicate(
+        files,
+        getPathFn,
+        (numDuplicatedFiles, path) ->
+            String.format(
+                "Detected %d %s files with the same path=%s in payload with commitId=%s for table=%s",
+                numDuplicatedFiles,
+                fileType,
+                path,
+                commitId.toString(),
+                tableIdentifier.toString()));
+  }
+
+  private <T> List<T> deduplicate(
```

Comment:
Hrm something I don't think I have to be worried about but open to feedback; this function may return files back in a different order from the order that they were received in. 

In the case of a pure AppendFiles operation, since we commit all datafiles in a single operation, changes in ordering shouldn't matter. 

In the case of a RowDelta operation, we still commit all datafiles/deletefiles in a single operation. I don't _think_  changes in ordering should matter here either but wouldn't mind confirmation from someone more familiar with this side.

Reply:
yeah I think you're good here, ordering of the calls within a single operation for RowDelta doesn't matter